### DISCUSSION TITLE
Are tags supported?

### DISCUSSION THREAD

**Discussion by ppazos** [spockframework/spock#1830] [General]
Is there a way to add tags to classes and/or methods so only the test classes/methods with certain tags are executed? 

We are trying to migrate tests from Robot Framework into Spock and this is a strong requirement for us, though not sure if Spock supports this.

Thanks.

**Comment by Vampire** [comment]
Yes, you can use `@Tag` which will also add according JUnit Platform tags which you can then for example include or exclude using Gradle or similar: https://spockframework.org/spock/docs/2.3/all_in_one.html#test-tag-extension

Additionally you can use the Spock configuration file to configure include and excludes based on arbitrary annotations, arbitrary base classes, which can also be calculated for example from some value given from outside like a Gralde build script: https://spockframework.org/spock/docs/2.3/all_in_one.html#_include_and_exclude

**Reply by ppazos** [reply]
@Vampire thanks, though I can't find how tests with specific tag/tags can be executed from the terminal, for instance I'm using Gradle, is there some documentation about that?

**Reply by Vampire** [reply]
There is no built-in way in Gradle. You control it from the build script, for example by having two tasks, one for one set, one for the other set and select the task or tasks to run from the commandline. Or you configure the task from a project property that you can then give from the command line is similar, but multiple tasks is better for up-to-date-ness and caching for example.

**Reply by ppazos** [reply]
I'll try it and get back.  Maybe controlling the executed tests by tag from Gradle could be a good feature request ?

**Reply by Vampire** [reply]
As I said, you can from Gradle build script.
